### ISSUE TITLE
Move JavaCSVTableSource operator from sql-api to the Java platform

### ISSUE THREAD

**Issue by zkaoudi** [apache/incubator-wayang#482]
There is a Java csv source operator inside the wayang sql api:
[JavaCSVTableSource](https://github.com/apache/incubator-wayang/blob/main/wayang-api/wayang-api-sql/src/main/java/org/apache/wayang/api/sql/sources/fs/JavaCSVTableSource.java).

We should move it to the wayang-platforms/wayang-java module, specifically [here](https://github.com/apache/incubator-wayang/tree/main/wayang-platforms/wayang-java/src/main/java/org/apache/wayang/java/operators) and rename it to JavaCSVFileSource because it is reading from a file and not a table in the database.

**Comment by kbeedkar** [comment]
Makes sense, the reason to call it JavaCSVTableSource was that it expects the csv fike to adhere to a certain format. Specifically, comma separated colname:type values. This is required to create an appropriate schema to query against.

**Comment by Chandanakt** [comment]
Hi @zkaoudi ,

My name is Chandana K T, and I’m a new aspiring contributor interested in getting involved with Apache Wayang. I’m planning to apply for Google Summer of Code (GSoC) and would like to gain hands-on experience with the codebase by taking up existing issues.

I’d be happy to work on this task. My understanding is to relocate JavaCSVTableSource from the SQL API module to the wayang-platforms/wayang-java module and rename it to JavaCSVFileSource, ensuring that the current functionality remains the same and that all usages are updated.

Please let me know if this sounds reasonable or if you have any guidance before I begin. Thanks!

**Comment by Chandanakt** [comment]
Hi @zkaoudi ,
I’ve opened a PR that moves JavaCSVTableSource from the SQL API module to the wayang-java platform module and renames it to JavaCSVFileSource.

The helper class CsvRowConverter was also moved accordingly, and all references were updated. No functional changes were introduced.

Looking forward to your feedback. Thanks!
